### ISSUE TITLE
machconf documentation inaccurate / incomplete

### ISSUE THREAD

**Issue by eirrgang** [radical-cybertools/radical.pilot#2588]
https://github.com/radical-cybertools/radical.pilot/blob/devel/docs/source/machconf.rst#writing-a-custom-resource-configuration-file
states that "All fields are mandatory, unless indicated otherwise below."

This does not seem to be true, with respect to the fields ``virtenv_mode``, ``virtenv``, and ``virtenv_dist``, all of which appear to be completely undocumented.

It looks like ``virtenv_mode`` may be required. ``virtenv`` may be required _iff_ ``virtenv_mode=="use"``?

**Comment by eirrgang** [comment]
It looks like these fields are owned by `radical.pilot.pmgr.launching.base.PMGRLaunchingComponent`. Many, if not all, of the other fields may be, too.

Specifically, the `_prepare_pilot()` method seems to be the direct consumer. If so, then all three of the `virtenv*` fields appear to be optional.

If there is not a convention for documenting schema, or if it is onerous to keep `machconf.rst` synchronized with `radical.pilot.pmgr.launching.base.PMGRLaunchingComponent._prepare_pilot()`, these local variables (and lines https://github.com/radical-cybertools/radical.pilot/blob/64253944063236c9850d5a8b35d2308bd5e78b31/src/radical/pilot/pmgr/launching/base.py#L561 through https://github.com/radical-cybertools/radical.pilot/blob/64253944063236c9850d5a8b35d2308bd5e78b31/src/radical/pilot/pmgr/launching/base.py#L593) could be moved to a simple container with documented members. Keeping the documentation at the definition, and the member initialization explicit would help keep new and changed fields documented.

**Comment by andre-merzky** [comment]
You are right, the resource config documentation is woefully out of data (see [here](https://github.com/radical-cybertools/radical.pilot/issues/2086)).  We are working on a restructure / revamp of the RP documentation in general, there should be significant changes over the next couple of months.

**Comment by eirrgang** [comment]
> You are right

Am I right just about the docs being out of date? Or can you confirm:
> `virtenv_mode` [is] required. `virtenv`  [is] required if-and-only-if `virtenv_mode=="use"`

**Comment by andre-merzky** [comment]
It is not required, by default the VE is updated.

**Comment by eirrgang** [comment]
all `virtenv*` fields are optional, `virtenv_mode` defaults to "update"

**Comment by andre-merzky** [comment]
This ticket collects items from several other tickets linked below.  The focus should be on defining a proper resource config class (based on `ru.TypedDict`), including the respective documentation for the supported settings.  I am going to close the other issues in favor of this one.

**Comment by eirrgang** [comment]
Regarding #2700: can we work on a specification for the `filesystem_endpoint` field? At this point in the conversation, my impression is that it is not meant to convey any path information, but just connection details. An attempt is made to normalize it to the root of the filesystem at the remote, but we don't know for sure where it will land. (right?) But we should be able to explain the normalization attempted.

Then, we should clarify how it is used. I.e. the meaning of `filesystem_endpoint` to the user amounts to how it affects file transfer destinations, I think.

**Comment by andre-merzky** [comment]
Yes - its named `endpoint` as in connection endpoint for that reason.

**Comment by eirrgang** [comment]
Assumed to be resolved by #2862
